channel/boss.go

```diff
@@ -0,0 +1,97 @@
+package channel
+
+import (
+	"log"
+	"sync/atomic"
+	"time"
+
+	"github.com/Hucaru/Valhalla/constant"
+)
+
+func manageSummonedBoss(inst *fieldInstance, mobID int32, server *Server) {
+	log.Println("Boss handler spawned for", mobID)
+
+	inst.dispatch <- func() {
+		inst.properties["eventActive"] = true
+	}
+
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+
+	timeout := time.NewTimer(time.Hour * 1)
+	defer timeout.Stop()
+
+	var finished atomic.Bool
+	finished.Store(false)
+
+	for {
+		select {
+		case <-ticker.C:
+			if finished.Load() {
+				return
+			}
+
+			inst.dispatch <- func() {
+				for _, mob := range inst.lifePool.mobs {
+					if mob.boss {
+						return
+					}
+				}
+
+				inst.properties["eventActive"] = false
+				finished.Store(true)
+			}
+		case <-timeout.C:
+			var returnMap int32
+
+			switch mobID {
+			case constant.MobPapalatusBall:
+				returnMap = constant.MapBossPapulatusReturn
+			case constant.MobZakum1Body:
+				returnMap = constant.MapBossZakumReturn
+			}
+
+			inst.dispatch <- func() {
+				field, ok := server.fields[returnMap]
+
+				if !ok {
+					log.Println("Error in getting field")
+					return
+				}
+
+				dest, err := field.getInstance(0)
+
+				if err != nil {
+					log.Println(err)
+					return
+				}
+
+				portal, err := dest.getPortalFromID(0)
+
+				if err != nil {
+					log.Println(err)
+					return
+				}
+
+				for _, plr := range inst.players {
+					server.warpPlayer(plr, field, portal, true)
+				}
+
+				inst.lifePool.eraseMobs()
```

Comment:
We need to reset the reactors too:
https://github.com/Hucaru/Valhalla/blob/master/channel/pools.go#L1165